### ISSUE TITLE
Programmatic Tab Display (Android)

### ISSUE THREAD

**Issue by andybbtt11** [wix/react-native-navigation#411]
I am unable to show the tabBar on Android by using the following:

```
this.props.navigator.toggleTabs({
    to: 'shown',
    animated: true
});
```

This works fine in iOS.  I am able to hide the tabs, but toggling them to show again does not work on Android.

**Comment by andybbtt11** [comment]
Just to clarify I tried excluding the to: property with no luck.

```
this.props.navigator.toggleTabs({
    animated: true
});
```

**Comment by guyca** [comment]
@andybbtt11 This seems to work fine in the [example project](https://github.com/wix/react-native-navigation/blob/master/example/src/screens/SecondTabScreen.js#L99).

```js
  onToggleTabsPress() {
    this.props.navigator.toggleTabs({
      to: this.tabsHidden ? 'shown' : 'hidden'
    });
    this.tabsHidden = !this.tabsHidden;
  }
```

**Comment by andybbtt11** [comment]
@guyca  You're absolutely right. I have verified that I can toggle the tabs on Android, however my situation is kind of odd as I need the tabBar to always be hidden on tabIndex: 0, but not the rest.

What I have tried is to set this on my screen which is used as tabIndex: 0

```js
// homescreen
static navigatorStyle = {
    tabBarHidden: true
};
```

and set this on the other screens (tabIndex 1-4)

```js
// other screens
static navigatorStyle = {
    tabBarHidden: false
};
```

Just by doing that, here are the results I get on iOS:
https://gfycat.com/JitteryPointedGrassspider
You see that it doesn't start out hidden but if I tab between them and go back, it'll hide then work properly. I was using the toggleTabs on didmount to initially hide it, so functionality was correct on iOS

Here are the results on Android:
https://gfycat.com/BiodegradableScarceIcelandgull

It starts off hidden but never comes back.

So yes, you're correct the toggleTabs functionality works on Android, I'm just in a weird situation and will keep trying different things, please let me know if you see any glaring errors/a better approach.

Thanks

**Comment by guyca** [comment]
Thanks for the detailed post. We didn't think someone would use `tabBarHidden: true` in a screen associated with a `BottomTab`. The property was added to hide/show bottom tabs when a screen is pushed/popped from the stack.

I'm interested to know, how the user navigates to other bottom tabs once he's in the first tab where tabs are hidden?

**Comment by andybbtt11** [comment]
I am using:
```js
this.props.navigator.switchToTab({
   tabIndex
});
```
to switch from the homeScreen.  Somehow iOS will then show/hide it like I wanted it to but Android does not.  On Android I can show the tabs again upon exiting the homeScreen and navigating to the next tab, but once I navigate back to the homeScreen, the tabBar will stay shown.
